### PR TITLE
Rustimprov

### PR DESCRIPTION
**Pull request by PhilipDaniels**
Hi as discussed last week in https://github.com/cs01/gdbgui/issues/113, here is a PR which I think improves the documentation around the Rust debugging experience. I have also updated the example to use the Cargo build system.

There is a screenshot in RUN.md: I have tried to anticipate the correct path after the merge.

This doesn't really close the issue, but I think it helps.

### CODE REVIEW

File: docs/RUN.md [hunk 1]
```diff
@@ -19,6 +19,47 @@ Now that gdbgui is open, you can interactively run a program with it.
...
+
+Instead of showing your `main` function the initial screen will be blank and `gdbgui` will print `File not found: main`.
+You need to help `gdbgui` out by typing `main` into the file browser box:
+
+![](https://raw.githubusercontent.com/cs01/gdbgui/master/screenshots/rust_main.png)
+
+and selecting the `main.rs` file. The source code should then appear in the browser and you can click to set breakpoints
+and run the program. Of course, if you want to break in some other file, you can find that in the file browser instead.
+
+The second niggle is not with `gdbgui` but with the GDB pretty-printing macros that Rust ships with. GDB can't find these by default,
```
Comment (cs01) [inline docs/RUN.md]: This implies the first has something to do with gdbgui, but I don't think it does. It has to do with gdb's response to the `break main` command.

gdbgui runs the gdb command `-break-insert main`, and then gets a breakpoint back from gdb. The breakpoint has a field called fullfile which is usually a path to a file, but when debugging rust, gdb cannot find the function and/or file and it just puts a breakpoint at an address.

File: docs/RUN.md [hunk 2]
```diff
@@ -19,6 +19,47 @@ Now that gdbgui is open, you can interactively run a program with it.
...
+cargo new myprog
+cd myprog
+cargo build
+```
+
+You can start debugging with a simple
+
+`gdbgui target/debug/myprog`
+
+There are a couple of small niggles.
```
Comment (cs01) [inline docs/RUN.md]: replace "niggles" with "difficulties"

**Comment by cs01** [comment]
@PhilipDaniels this is great, thanks for doing this.

Since there is no makefile to run a single make command that builds a program and launches gdbgui like the other examples, can you make a shell script that calls the rust command then launches gdbgui?

Also can a README.md be added that says to run that script, and describe (in one small sentence) what will happen when it's run?

**Comment by PhilipDaniels** [comment]
@cs01 Added one commit that I think addresses all your comments.

File: docs/RUN.md [hunk 3]
```diff
@@ -19,6 +19,47 @@ Now that gdbgui is open, you can interactively run a program with it.
...
+cargo build
+```
+
+You can start debugging with a simple
+
+`gdbgui target/debug/myprog`
+
+There are a couple of small difficulties.
+
+Instead of showing your `main` function the initial screen will be blank and `gdbgui` will print `File not found: main`.
```
Comment (cs01) [inline docs/RUN.md]: I tested this myself and found a couple things.

1.) this `file not found` error is sort of expected, and can be worked around in a couple ways. See https://stackoverflow.com/q/24664994/2893090.

2.) when I build this on my mac, I get a bunch of warnings like this

> warning /Users/chad/examples/rust/target/debug/deps/hello-486956f9dde465e5.9elsx31vb4it187.rcgu.o': can't open to read symbols: No such file or directory.

It's true, those files don't exist. But `rust/target/debug/deps/hello-486956f9dde465e5` does. So if I run `symbol-file /Users/cssmith/git/gdbgui/examples/rust/target/debug/deps/hello-486956f9dde465e5`, gdb successfully loads the symbol table and `hello::main` successfully inserts a breakpoint and opens the src/main.rs file.

**Comment by PhilipDaniels** [comment]
Thanks for merging this.

**Pull request merged**